File: src/main/java/pl/dolien/shop/auth/registration/RegistrationService.java
package pl.dolien.shop.auth.registration;

import pl.dolien.shop.auth.registration.dto.RegistrationDTO;
import pl.dolien.shop.user.User;

import javax.management.relation.RoleNotFoundException;

public interface RegistrationService {

    User registerUser(RegistrationDTO dto) throws RoleNotFoundException;
}

File: shop-ui/src/app/auth/registration/services/registration/registration.service.ts
import { Injectable } from '@angular/core';
import { environment } from '../../../../../environments/environment.development';
import { HttpClient } from '@angular/common/http';
import { RegistrationDTO } from '../../models/registration-dto/registration-dto';
import { Observable } from 'rxjs';
import { User } from '../../../../user/models/user/user';

@Injectable({
  providedIn: 'root',
})
export class RegistrationService {
  private readonly registerUrl = `${environment.url}/auth/register`;

  constructor(private httpClient: HttpClient) {}

  register(registrationDTO: RegistrationDTO): Observable<User> {
    return this.httpClient.post<User>(this.registerUrl, registrationDTO);
  }
}
